### PR TITLE
Adds noise to EMCal L1 Trigger set via SimParams

### PR DESCRIPTION
**Pull request by moliver813**
Updates SimParams, adding L1ADCNoise param
Adds noise to L1 trigger timesums based on param in SimParam OCDB

### CODE REVIEW

**Comment by gconesab** [comment]
Hi, I leave this to Markus, but why have you removed the copy constructors? I am not sure they are ever needed but they do not harm.

**Comment by mfasDa** [comment]
@gconesab Singleton classes must never have a copy constructor - they violate the idea behind a singleton that there can never be two instances at the same time.

File: EMCAL/EMCALsim/CMakeLists.txt
Comment (mfasDa) [inline EMCAL/EMCALsim/CMakeLists.txt:26]: Eventually we need to move the constants file to EMCALutils, not sure about the dependency here. I would prefer to have the dependencies as lightweight as possible.

File: EMCAL/EMCALbase/AliEMCALSimParam.cxx
Comment (mfasDa) [inline EMCAL/EMCALbase/AliEMCALSimParam.cxx:140]: Please rather implement operator<< for AliEMCALSimParams - we are in the c++ time and the time for such print functions are over. We usually do the following:

- Add PrintStream functions (in header: include iosfwd instead of iostream) with the following interface
``` c++
void PrintStream(std::ostream &stream) const;
```
  and implement the printing in the same way as if one would use std::cout.
- Implement operator<< becomes now trivial:
``` c++
std::ostream &operator<<(std::ostream &stream, const AliEMCALSimParam &param) {   
  param.PrintStream(stream);
  return stream;
}
```
- Finally the print function becomes:
``` c++ 
void AliEMCALSimParam::Print(Option_t *) {
  std::cout << *this << std::endl;
}
```

This adds more flexibility to the debug streaming as besides the print function this can be used in std::cout, but better with any kind of logging library like the loggers in AliLog (AliErrorStream, AliWarningStream, AliInfoStream, AliDebugStream) ... where log levels can be set

**Review by mfasDa — CHANGES_REQUESTED** [review]
Please have a look at the logging, and eventually we need to fix the dependency chain here.

File: EMCAL/EMCALbase/AliEMCALSimParam.cxx
Comment (moliver813) [reply EMCAL/EMCALbase/AliEMCALSimParam.cxx:140]: For the Print(Option_t *) function, should it be AliInfoStream() or AliDebugStream() instead of std::cout?

Comment (mfasDa) [reply EMCAL/EMCALbase/AliEMCALSimParam.cxx]: This is basically a compatibility function to ROOT5 style logging where I think std::cout is the way to go.

**Comment by mfasDa** [comment]
+1

**Comment by ktf** [comment]
+1

I think it's because a CMakeLists is modified, but I am not 100% sure.

**Comment by mfasDa** [comment]
@ktf But these CMankeLists within EMCAL, so we cannot even update our own CMakeLists any more?

**Comment by moliver813** [comment]
I just added #include <iostream> to the AliEMCALSimParams.h. I think the absence of that was causing the build issue in the MacOS build.

**Comment by mfasDa** [comment]
+1

**Comment by mfasDa** [comment]
The approval should be withdrawn for the moment, as AliEMCALSimParams.h now includes both iostream and iosfwd, however it should only include iosfwd and must under no circumstance include iostream. Waiting for a fix before the PR can be merged.

**Comment by moliver813** [comment]
Just submitted a fix that moves the iostream include to AliEMCALSimParam.cxx

**Comment by mfasDa** [comment]
+1 

Thanks @moliver813 for fixing! Should be fine now.

@ktf Could you please approve once more?

**Comment by ktf** [comment]
+1

**Pull request merged**
